File: frontend/src/app/person.ts
export class Person {
  public id?: number;
  public lastname!: string;
  public firstname!: string;
  public minor?: boolean;
  public address1?: string;
  public address2?: string;
  public height?: number;
  public weight?: number;
  public birthdate!: Date;

  constructor(person: Person) {
    this.id = person.id;
    this.lastname = person.lastname;
    this.firstname = person.firstname;
    this.minor = person.minor;
    this.address1 = person.address1;
    this.address2 = person.address2;
    this.height = person.height;
    this.weight = person.weight;
    this.birthdate = person.birthdate;
  }
}

File: backend/src/main/java/fr/pif/formationtests/model/Person.java
package fr.pif.formationtests.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Person {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String lastname;
    @Column(nullable = false)
    private String firstname;

    @Column(columnDefinition = "boolean default false")
    private Boolean minor = false;

    @Column(length = 10)
    private String address1;
    @Column(length = 100)
    private String address2;

    private Double weight;

    private Double height;

    @Column(nullable = false)
    private LocalDate birthdate;
}
